File: golang/118.pascals-triangle.go
func generate(numRows int) [][]int {
	pas := make([][]int, numRows)

	for i := range numRows {
		pas[i] = make([]int, i+1)
		pas[i][0] = 1
		pas[i][i] = 1

		for j := 1; j < i; j++ {
			pas[i][j] = pas[i-1][j-1] + pas[i-1][j]
		}
	}

	return pas
}

File: python/118.pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        pas = [[] for _ in range(numRows)]

        for i in range(numRows):
            pas[i] = [1] * (i + 1)
            for j in range(1, i):
                pas[i][j] = pas[i - 1][j - 1] + pas[i - 1][j]

        return pas

